hudi-client/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -562,4 +566,28 @@ public void validateInsertSchema() throws HoodieInsertException {
       throw new HoodieInsertException("Failed insert schema compability check.", e);
     }
   }
+
+  public HoodieFileFormat getBaseFileFormat() {
+    return metaClient.getTableConfig().getBaseFileFormat();
+  }
+
+  public HoodieFileFormat getLogFileFormat() {
+    return metaClient.getTableConfig().getLogFileFormat();
+  }
+
+  public HoodieLogBlockType getLogDataBlockFormat() {
+    switch (getBaseFileFormat()) {
+      case PARQUET:
+        return HoodieLogBlockType.AVRO_DATA_BLOCK;
+      case HFILE:
+        return HoodieLogBlockType.HFILE_DATA_BLOCK;
+      default:
+        throw new HoodieException("Base file format " + getBaseFileFormat()
+            + " does not have associated log block format");
+    }
+  }
+
+  public boolean requireSortedRecords() {
```

Comment:
Instead of this, lets return merge algorithm to be employed.

Reply:
See my comments above. I have created a new merge handle.